### PR TITLE
Fixed RNN for TF 1.0

### PR DESCRIPTION
**Pull request by hunkim**
이번 lab강의 재녹음준비로 RNN을 다시 작성해보았습니다. 수업공개하기전에 코드 전문가에게 조언을 받기 위한 PR입니다.

RNN에 들어가는 숫자들이 많아서 지난번에 좀 제가 햇갈린 부분이 있었는데 바로 잡았습니다.

그리고 RNN만으로는 다음 문자 정확한 예측이 어려운데 뒷단에 softmax 하나 추가해보았습니다. (30번째줄 넣고 빼고 큰 차이가 있습니다.)

그리고 변수는 가급적 같은 의미일경우 재사용하여 의미 전달을 위해 노력. 이게 초보자들에게는 더 햇갈릴수도 있을듯 합니다.
(예: outputs = tf.reshape(outputs, [-1, dic_size]))

보시고 의견주시면 수정후 수업에 반영하겠습니다.

### CODE REVIEW

**Comment by golbin** [comment]
전문가라고 하기엔 저도 초보자라..^^;

전반적으로 네이밍이 명확해져 이해하기 좋아진 것 같습니다.

다만, 제가 주석으로 각 행렬이나 텐서들의 쉐이프나 결과값들을 넣어뒀었는데요. 이게 초보자 입장에서는 눈에 안보이니까 이해하기가 어렵더라고요 ^^; 주석으로 넣지는 않아도 눈에 보이게 설명해 주시면 좋을 것 같습니다.

그리고, 1.0 기준으로 재작성하신 코드라면, loss 함수로 tf.contrib.legacy_seq2seq.sequence_loss_by_example 대신에 tf.contrib.seq2seq.sequence_loss 를 사용하면 어떨까 합니다.

tf.contrib.legacy_seq2seq 이 모듈은 Deprecated 되어서 이참에 tf.contrib.seq2seq 를 소개하고 사용하는 것이 좋을 것 같아요.

**Comment by hunkim** [comment]
역시 전문가. 새로운 api들이 많이 생겨났군요.

batch 개념이 들어와서 result부분 좀 복잡해지네요. lstm의 파워를 보여주기 위해 그낭 softmx 로만 구현한 버전: 02_xx 는 동작되지 않음을 보여줄 예정.

말씀처럼 강의에는 jupiter를 사용해서 vector 사이즈 같은거 보여줘야 햘듯합니다.

참고로   https://github.com/tensorflow/models/blob/master/tutorials/rnn/ptb/ptb_word_lm.py 도 PR 보내 주어야 할듯. :-)
